src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+    }
+
+    if (!animal.health) {
+      Animal.alive = Animal.alive.filter((subject) => subject.health !== 0);
```

Comment:
[CODE PERFORMANCE] Instead of checking for `!animal.health` which is a bit unclear and can lead to bugs (since 0 is falsy but so is `undefined` and `null`), it's better to explicitly check if the animal's health is less than or equal to 0 before filtering. Moreover, the filtering condition should check for `subject.health > 0` to keep alive animals with health above 0.